crates/sdk/src/env.rs

```diff
@@ -181,3 +184,47 @@ pub fn state_write<T: AppState>(state: &T) {
     };
     let _ = storage_write(STATE_KEY, &data);
 }
+
+/// Generate a new random UUID v4.
+#[inline]
+#[must_use]
+pub fn generate_uuid() -> Uuid {
+    #[cfg(target_arch = "wasm32")]
+    {
+        unsafe { sys::generate_uuid(DATA_REGISTER) };
+        Uuid::from_slice(&read_register_sized::<16>(DATA_REGISTER).expect("Must have UUID"))
+            .expect("UUID must be valid")
+    }
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        Uuid::new_v4()
+    }
+}
+
+/// Gets the current time.
+#[inline]
+#[must_use]
+pub fn time_now() -> u64 {
+    #[cfg(target_arch = "wasm32")]
+    {
+        unsafe { sys::time_now(DATA_REGISTER) };
+        u64::from_le_bytes(
+            read_register(DATA_REGISTER)
+                .expect("Must have time")
+                .try_into()
+                .expect("Invalid time bytes"),
+        )
+    }
+    #[cfg(not(target_arch = "wasm32"))]
+    #[expect(
+        clippy::cast_possible_truncation,
+        reason = "Impossible to overflow in normal circumstances"
+    )]
+    #[expect(clippy::expect_used, reason = "Effectively infallible here")]
+    {
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards to before the Unix epoch!")
+            .as_nanos() as u64
+    }
```

Comment:
same here

Reply:
So I think the purpose here has been misunderstood... and the premise seems incorrect. If I remove this, then the build script no longer works:

```
error[E0432]: unresolved import `calimero_sdk::env::time_now`
   --> crates/storage/src/entities.rs:219:5
    |
219 | use calimero_sdk::env::time_now;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `time_now` in `env`
    |
note: found an item that was configured out
   --> /Users/dan/Code/calimero/crates/sdk/src/env.rs:233:8
    |
233 | pub fn time_now() -> u64 {
    |        ^^^^^^^^
note: the item is gated behind the `wasm32` feature
   --> /Users/dan/Code/calimero/crates/sdk/src/env.rs:230:7
    |
230 | #[cfg(target_arch = "wasm32")]
    |       ^^^^^^^^^^^^^^^^^^^^^^
```

It may be possible to do what you're suggesting in a different way, but I'm not sure how, and what's in place currently works.

Reply:
I'm confused, the point of this comment is to highlight that feature gating for wasm32 is unnecessary.

Reply:
Yes, I understand the point of the comment, but as illustrated above, that's incorrect. If you attempt what you have described, then the build fails. Therefore this is necessary.

Reply:
The build fails because you still have a feature flag 🤷🏽‍♂️

Reply:
That's the point, isn't it? In order to build, this needs to be available 🤔

Reply:
and that's the point, it doesn't need it, you don't need to accommodate for anything but wasm

Reply:
For whatever reason, if that is removed, the build script fails.

Reply:
what does it look like when it's removed?

Reply:
You know more about this side than I do - I've only recently started fiddling with it, in order to try to test this stuff. I wrote what I did in order for it to fully work. Removing what you've suggested breaks it in the way described. I'd say the best path forward at this stage is for you to make the changes that you are thinking of, as it may be that I've misunderstood you.